sources/packages/backend/apps/api/src/route-controllers/note/_tests_/e2e/note.institutions.controller.getStudentNotes.e2e-spec.ts

```diff
@@ -0,0 +1,238 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  createE2EDataSources,
+  createFakeInstitutionLocation,
+  createFakeUser,
+  E2EDataSources,
+  saveFakeApplication,
+  saveFakeStudentRestriction,
+} from "@sims/test-utils";
+import {
+  InstitutionLocation,
+  Note,
+  NoteType,
+  RestrictionNotificationType,
+} from "@sims/sims-db";
+import {
+  createFakeNote,
+  saveFakeStudentNotes,
+} from "@sims/test-utils/factories/note";
+import {
+  authorizeUserTokenForLocation,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAuthRelatedEntities,
+  getInstitutionToken,
+  InstitutionTokenTypes,
+} from "../../../../testHelpers";
+
+describe("NoteInstitutionsController(e2e)-getStudentNotes", () => {
+  let app: INestApplication;
+  let collegeFLocation: InstitutionLocation;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+
+    // College F.
+    const { institution: collegeF } = await getAuthRelatedEntities(
+      dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeFLocation = createFakeInstitutionLocation(collegeF);
+    await authorizeUserTokenForLocation(
+      dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+  });
+
+  it("Should get all note types for a student when student they are available.", async () => {
+    // Arrange
+    // Create new application.
+    const savedApplication = await saveFakeApplication(db.dataSource, {
+      institutionLocation: collegeFLocation,
+    });
+
+    const student = savedApplication.student;
+    const [
+      generalNotes,
+      systemNotes,
+      applicationNotes,
+      designationNotes,
+      overawardNotes,
+      programNotes,
+      restrictionNotes,
+    ] = await saveFakeStudentNotes(
```

Comment:
As a suggestion, the notes can be created as below.
```ts
const fakeNotes = Object.values(NoteType).map((nodeType) =>
  createFakeNote(nodeType),
);
const savedNotes = await saveFakeStudentNotes(
  db.dataSource,
  fakeNotes,
  student.id,
);
```
Then the expected return can also be checked as below.
```ts
const expectedAPIReturnNotes = savesNotes.map((savedNote) =>
  noteToApiReturn(savedNote),
);
```